### ISSUE TITLE
JSON macro fails in some cases

### ISSUE THREAD

**Issue by 89z** [w-henderson/Humphrey#102]
these all work fine:

~~~rust
use humphrey_json::json;

fn main() {
   let v = json!(true);
   dbg!(v);
   let v = json!(1234);
   dbg!(v);
   let v = json!("Hello, world!");
   dbg!(v);
   let v = json!({"key":"value"});
   dbg!(v);
}
~~~

but for some reason, fully qualified fails in one case:

~~~rust
fn main() {
   let v = humphrey_json::json!(true);
   dbg!(v);
   let v = humphrey_json::json!(1234);
   dbg!(v);
   let v = humphrey_json::json!("Hello, world!");
   dbg!(v);
   // cannot find macro `json` in this scope
   // let v = humphrey_json::json!({"key":"value"});
   // dbg!(v);
}
~~~

**Comment by w-henderson** [comment]
This happens because the macro is supposed to be imported with the prelude, as is shown in the demonstration examples:

```rs
use humphrey_json::prelude::*;
```

This brings all the necessary traits into scope and you'll run into issues down the line if these traits are not in scope, as the macro won't be able to convert certain types to and from JSON.

However, for simple cases such as these, I think it should work as-is so I'm working on a fix at the moment.

**Comment by 89z** [comment]
much appreciated!

**Comment by w-henderson** [comment]
I ended up rewriting the macro from scratch because there were a number of other problems with the implementation that I'd been meaning to fix for ages. I've released all these bug fixes in v0.2.2.

**Comment by 89z** [comment]
confirmed fixed, thank you!
